### PR TITLE
feat: Sum and Average aggregations

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java [hunk 1]
```diff
@@ -0,0 +1,856 @@
...
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ITQueryAggregationsTest {
+  private Firestore firestore;
+
+  @Before
+  public void setUpFirestore() {
+    // TODO: stop using emulator for these tests once prod is ready.
+    firestore = FirestoreOptions.newBuilder().setHost("localhost:8080").build().getService();
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java:32]: I think this line needs to be updated for production launch.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateField.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class AggregateField {
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateField.java:23]: javadoc

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java [hunk 3]
```diff
@@ -0,0 +1,856 @@
...
+                "metadata",
+                map("pages", 50, "rating", map("critic", 4, "user", 4))));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot =
+        collection
+            .aggregate(
+                sum("metadata.pages"),
+                average("metadata.pages"),
+                average("metadata.rating.critic"),
+                sum("metadata.rating.user"),
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java:518]: nit: This test could run against prod and provide value by only running against "metadata.pages".

**Review by MarkDuckworth — COMMENTED** [review]
Javadoc needs to be added for AggregateField and there's a test config still pointing to the emulator. Otherwise, it looks good to me.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateField.java (hunk 2 above)
Comment (ehsannas) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateField.java:24]: Done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java (hunk 1 above)
Comment (ehsannas) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java]: Good catch. Fixed.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java (hunk 3 above)
Comment (ehsannas) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java]: Fixed.

**Review by MarkDuckworth — APPROVED** [review]
LGTM. Can land as in, but as you mentioned, we could add additional return type info to javadoc on the AggregateField class to improve this. If there are further updates, I'm happy to take a look.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java [hunk 4]
```diff
@@ -25,32 +25,27 @@
 import com.google.api.gax.rpc.StreamController;
 import com.google.cloud.Timestamp;
 import com.google.cloud.firestore.v1.FirestoreSettings;
-import com.google.firestore.v1.RunAggregationQueryRequest;
-import com.google.firestore.v1.RunAggregationQueryResponse;
-import com.google.firestore.v1.RunQueryRequest;
-import com.google.firestore.v1.StructuredAggregationQuery;
-import com.google.firestore.v1.Value;
+import com.google.firestore.v1.*;
```
Comment (tom-andersen) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java:28]: Using `*` wildcard goes against style guide.

go/java-style#s3.3.1-wildcard-imports

**Review by markarndt — APPROVED** [review]

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java (hunk 4 above)
Comment (ehsannas) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java]: Thanks for catching this @tom-andersen . I believe my IDE has done this behind my back! 🙅
Fixed.

**Review by markarndt — APPROVED** [review]

**Review by MarkDuckworth — APPROVED** [review]
LGTM

**Review by MarkDuckworth — APPROVED** [review]
LGTM

**Pull request merged**